### ISSUE TITLE
PreLab 3: Απεικόνιση και Βήμα 5

### ISSUE THREAD

**Issue by you-know-who-lab** [slp-ntua/patreco-lab-support-19-20#26]
Καλησπέρα,

1) Σχετικά με τις απεικονίσεις των φασματογραφημάτων, εισάγοντας στην librosa.display.specshow την παράμετρο x_axis='time', στα μη beat-synced εμφανίζεται (max) χρόνος πχ 30 ενώ στα αντίστοιχα beat-synced εμφανίζεται 2. Αυτό φαναζομαι δεν είναι σωστό, ή είναι; Μπορώ να κάνω κάτι;

2) Τι αποτελέσματα περιμένουμε από την ταξινόμηση; Κατά την εκπαίδευση το loss ξεκινάει από το 2.3 και δεν πέφτει κάτω από 2.15 (μετά από 22 εποχές) και αυτό με προβληματίζει. Αντίστοιχα αποτελέσματα παίρνω κι αν εκπαιδεύσω το δίκτυο σε ένα μόνο batch όπως προτείνεται για έλεγχο. Υπάρχει περίπτωση το LSTM ενώ ήταν αρκετά καλό στην δεύτερη άσκηση, να μην "κάνει" για εδώ; Πρέπει να το κάνω πιο περίπλοκο;

Ευχαριστώ

**Comment by shmpe** [comment]
> 2\. Αντίστοιχα αποτελέσματα παίρνω κι αν εκπαιδεύσω το δίκτυο σε ένα μόνο batch όπως προτείνεται για έλεγχο.

δοκίμασε να χρησιμοποιήσεις learning rate κοντά στο 0.0001

**Comment by lobracost** [comment]
Αντιμετωπίζω και εγώ το ίδιο πρόβλημα στο LSTM. Συγκεκριμένα έχω δοκιμάσει διάφορες τιμές για το learning rate προσθέτωντας και έναν learning rate scheduler. Έχω πειραματιστεί με dropout, num_layers και άλλες μεταβλητές, αλλά τελικά η καλύτερη επίδοση που πετυχαίνω είναι 13% validation accuracy και 1.3 training loss. Μήπως υπάρχει κάτι που μου διαφεύγει;

**Comment by georgepar** [comment]
> 1. Σχετικά με τις απεικονίσεις των φασματογραφημάτων, εισάγοντας στην librosa.display.specshow την παράμετρο x_axis='time', στα μη beat-synced εμφανίζεται (max) χρόνος πχ 30 ενώ στα αντίστοιχα beat-synced εμφανίζεται 2. Αυτό φαναζομαι δεν είναι σωστό, ή είναι; Μπορώ να κάνω κάτι;

Αυτο δεν είναι σωστο. Χρησιμοποιήστε x_axis='linear' για να σας δείχνει αριθμό δειγμάτων.

> 2. Τι αποτελέσματα περιμένουμε από την ταξινόμηση; Κατά την εκπαίδευση το loss ξεκινάει από το 2.3 και δεν πέφτει κάτω από 2.15 (μετά από 22 εποχές) και αυτό με προβληματίζει. Αντίστοιχα αποτελέσματα παίρνω κι αν εκπαιδεύσω το δίκτυο σε ένα μόνο batch όπως προτείνεται για έλεγχο. Υπάρχει περίπτωση το LSTM ενώ ήταν αρκετά καλό στην δεύτερη άσκηση, να μην "κάνει" για εδώ; Πρέπει να το κάνω πιο περίπλοκο;

Το LSTM δε θα δώσει ιδιαίτερα καλά αποτελέσματα. Υπολογίστε γύρω στο 30% accuracy
